src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1114,113 +1106,192 @@ public String getFileName()
         }
 
         /**
-         * Creates an {@link net.dv8tion.jda.api.entities.Icon Icon} instance for
-         * this attachment if {@link #isImage() isImage()} is {@code true}!
-         * <br>This is a convenience method that can be used to retrieve an Icon from an attachment image link which
-         * requires a set user-agent to be loaded.
+         * Enqueues a request to retrieve the contents of this Attachment.
+         * <br>The receiver is expected to close the retrieved {@link java.io.InputStream}.
          *
-         * <p>When a global proxy was specified via {@link net.dv8tion.jda.api.JDABuilder JDABuilder} this will use the
-         * specified proxy to create an {@link java.io.InputStream InputStream} otherwise it will use a normal {@link java.net.URLConnection URLConnection}
-         * with the User-Agent for the currently logged in account.
+         * <h2>Example</h2>
+         * <pre>{@code
+         * public void printContents(Message.Attachment attachment)
+         * {
+         *     attachment.retrieveData().thenAccept(in -> {
+         *         StringBuilder builder = new StringBuilder();
+         *         byte[] buf = byte[1024];
+         *         int count = 0;
+         *         while ((count = in.read(buf)) > 0)
+         *         {
+         *             builder.append(new String(buf, 0, count));
+         *         }
+         *         in.close();
+         *         System.out.println(builder);
+         *     }).exceptionally(t -> { // handle failure
+         *         t.printStackTrace();
+         *         return null;
+         *     });
+         * }
+         * }</pre>
          *
-         * @throws IOException
-         *         If an IOError occurs while reading the image
-         * @throws java.lang.IllegalStateException
-         *         If this is not an image attachment
-         *
-         * @return {@link net.dv8tion.jda.api.entities.Icon Icon} for this image attachment
-         *
-         * @since  3.4.0
+         * @return {@link java.util.concurrent.CompletableFuture} - Type: {@link java.io.InputStream}
          */
-        public Icon getAsIcon() throws IOException
+        public CompletableFuture<InputStream> retrieveData() // it is expected that the response is closed by the callback!
```

Comment:
Thoughts on changing this to `retrieveInputStream`?

Reply:
I'm indifferent about it